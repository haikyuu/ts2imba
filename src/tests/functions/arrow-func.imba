import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code4 = '''
let x = (i, j) => i+j
'''

let imba-code4 = '''
let x = do(i, j) i + j

'''
test 'two params' do
	const result = await build tsx-code4
	expect(result.code).toEqual(imba-code4)
	const out = imbac.compile(result.code, sourceId: 'sth')


let tsx-code1 = '''
let App = () => {
	let a = 34
	return a
}
'''

let imba-code1 = '''
let App = do
	let a = 34
	a

'''
test 'arrow functions ' do
	const result = await build tsx-code1
	expect(result.code).toEqual(imba-code1)
	const out = imbac.compile(result.code, sourceId: 'sth')

let tsx-code = '''
let App = () => {
	return (
		<div>
			<span>Test</span>
		</div>
	)
}
'''

let imba-code = '''
tag App
	def render
		<self>
			<div>
				<span>
					"Test"

'''
test 'arrow functions component' do
	const result = await build tsx-code
	expect(result.code).toEqual(imba-code)
	const out = imbac.compile(result.code, sourceId: 'sth')

let tsx-code3 = '''
const b64toBlob = (b64Data, contentType='', sliceSize=512) => {
  const byteCharacters = atob(b64Data);
}
'''

let imba-code3 = '''
const b64toBlob = do(b64Data, contentType = "", sliceSize = 512)
	const byteCharacters = atob(b64Data)
	return

'''
test 'assignment pattern' do
	const result = await build tsx-code3
	console.log result.code
	expect(result.code).toEqual(imba-code3)
	const out = imbac.compile(result.code, sourceId: 'sth')
