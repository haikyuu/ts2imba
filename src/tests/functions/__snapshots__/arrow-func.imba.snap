// Vitest Snapshot v1

exports[`arrow functions 1223 1`] = `
"let App = do
	let a = 34
	return a
"
`;

exports[`arrow functions component 1`] = `
"tag App
	def render
		<self>
			<div>
				<span>
					\\"Test\\"
"
`;

exports[`assignment pattern 23 1`] = `
"const b64toBlob = do(b64Data, contentType = \\"\\", sliceSize = 512)
	const byteCharacters = atob(b64Data)
"
`;

exports[`two params 33 1`] = `
"let x = do(i, j) i + j
"
`;
